commit 723a8c3caf6fefd19c6b172426a4e872b2d0afdf
Author: Ian Campbell <ian.campbell@citrix.com>
Date:   Wed Sep 30 11:57:22 2009 +0100

    xenserver: Disable brcompatd
    
    The compat layer is now unused in the vswitch branch of XenServer.

diff --git a/xenserver/etc_init.d_vswitch b/xenserver/etc_init.d_vswitch
index e63a001..0857f89 100755
--- a/xenserver/etc_init.d_vswitch
+++ b/xenserver/etc_init.d_vswitch
@@ -24,8 +24,8 @@
 test -e /etc/sysconfig/vswitch && . /etc/sysconfig/vswitch
 
 # General config variables in /etc/sysconfig/vswitch
-: ${ENABLE_BRCOMPAT:=y}
-: ${ENABLE_FAKE_PROC_NET:=y}
+: ${ENABLE_BRCOMPAT:=n}
+: ${ENABLE_FAKE_PROC_NET:=n}
 : ${FORCE_COREFILES:=y}
 
 # Config variables specific to ovs-vswitchd
@@ -89,7 +89,7 @@ function insert_modules_if_required {
         action "Inserting llc module" modprobe llc
         action "Inserting openvswitch module" modprobe openvswitch_mod
     fi
-    if [ -n "$BRCOMPATD_PIDFILE" ] && ! lsmod | grep -q "brcompat_mod"; then
+    if [ "$ENABLE_BRCOMPATD" = "y" ] && [ -n "$BRCOMPATD_PIDFILE" ] && ! lsmod | grep -q "brcompat_mod"; then
         action "Inserting brcompat module" modprobe brcompat_mod
     fi
 }
@@ -267,7 +267,9 @@ function start {
     fi
 
     start_vswitchd
-    start_brcompatd
+    if [ "${ENABLE_BRCOMPATD}" = "y" ] ; then
+	start_brcompatd
+    fi
     reload_vswitchd  # ensures ovs-vswitchd has fully read config file.
     touch /var/lock/subsys/vswitch
 }
diff --git a/xenserver/usr_share_vswitch_scripts_sysconfig.template b/xenserver/usr_share_vswitch_scripts_sysconfig.template
index 3209477..2b3f8e7 100644
--- a/xenserver/usr_share_vswitch_scripts_sysconfig.template
+++ b/xenserver/usr_share_vswitch_scripts_sysconfig.template
@@ -9,13 +9,13 @@
 
 # ENABLE_BRCOMPAT: If 'y' than emulate linux bridging interfaces
 #    using the brcompat kernel module and ovs-brcompatd daemon
-# ENABLE_BRCOMPAT=y
+# ENABLE_BRCOMPAT=n
 
 # ENABLE_FAKE_PROC_NET: If 'y' then emulate linux bonding and vlan
 #    files in /proc as if the bonding and vlan demultiplexing done in
 #    ovs-vswitchd were being implemented using existing Linux mechanisms.
 #    This is useful in some cases when replacing existing solutions.
-# ENABLE_FAKE_PROC_NET=y
+# ENABLE_FAKE_PROC_NET=n
 
 # FORCE_COREFILES: If 'y' then core files will be enabled.
 # FORCE_COREFILES=y
